src/actionCanRead.h

```diff
@@ -0,0 +1,57 @@
+/******************************************************************************
+ *                                                                            *
+ * Copyright (C) 2019 Fondazione Istituto Italiano di Tecnologia (IIT)        *
+ * All Rights Reserved.                                                       *
+ *                                                                            *
+ ******************************************************************************/
+
+ /**
+  * @file ActionCanRead.h
+  * @author Davide Tomé <davide.tome@iit.it>
+  */
+
+#pragma once
+
+#include <actionYarp.h>
+#include <yarp/dev/PolyDriver.h>
+#include <yarp/dev/CanBusInterface.h>
+#include <yarp/sig/Vector.h>
+
+
+using namespace yarp::dev;
+using namespace yarp::sig;
+using namespace yarp::os;
+using namespace yarp;
+
+namespace YarpActions
+{
+
+class ActionCanRead : public ActionYarp
+{
+  
+
+public:
+    ActionCanRead(const CommandAttributes& commandAttributes, const std::string& testCode);
+    execution execute(const TestRepetitions& testrepetition) override;
+    void beforeExecute() override;
+
+protected:
+  PolyDriver driver;
```

Comment:
For member variabiles use the following convention
camelcasename_ note the final _